### PR TITLE
[DE] Re-implementing strict double validation

### PR DESCRIPTION
**Pull request by rinkk**
Previous implementation didn't work with small value ranges for some reason (such as GMSH preferences where some params need to be in [0;1]).

This fixes an issue with unchangable parameter values in the GMSH dialog.
Also adjusts the minimum default value for parameters to something more reasonable.

(No entry to the changelog because this was supposed to work anyway. I *think* it *did* work in the past, possibly something was changed in QDoubleValidator?)

### CODE REVIEW

File: Applications/DataExplorer/DataView/GMSHPrefsDialog.cpp [hunk 1]
```diff
@@ -147,16 +147,17 @@ void GMSHPrefsDialog::accept()
...
+        double const min_scaling_factor (1e-10);
         max_number_of_points_in_quadtree_leaf = BaseLib::str2number<unsigned> (
                 param1->text().toStdString().c_str());
         if (max_number_of_points_in_quadtree_leaf == 0)
             max_number_of_points_in_quadtree_leaf = 10;
         mesh_density_scaling_pnts = fabs (param2->text().toDouble());
-        if (mesh_density_scaling_pnts < sqrt(std::numeric_limits<double>::epsilon()))
-            mesh_density_scaling_pnts = 0.5;
+        if (mesh_density_scaling_pnts < min_scaling_factor)
+            mesh_density_scaling_pnts = min_scaling_factor;
```
Comment (TomFischer) [inline Applications/DataExplorer/DataView/GMSHPrefsDialog.cpp]: In the previous version the value was 10 x larger than the value for `mesh_density_scaling_stations`. Now, `mesh_density_scaling_pnts` equals `mesh_density_scaling_stations`. Is this change intended?

Comment (rinkk) [reply Applications/DataExplorer/DataView/GMSHPrefsDialog.cpp]: Yes, it is intended. The smallest possible number allowed by the DoubleValidator is "0", but the algorithm requires a small positive value. So instead of setting it to the (relatively large) default value, it is now set to the previously allowed minimum value of 1^{-10}.
